Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -11,8 +11,57 @@
 import Foundation
 import SymbolKit
 
+typealias OverloadDeclaration = (
+    declaration: [SymbolGraph.Symbol.DeclarationFragments.Fragment],
+    reference: ResolvedTopicReference
+)
+
 /// Translates a symbol's declaration into a render node's Declarations section.
 struct DeclarationsSectionTranslator: RenderSectionTranslator {
+    /// A mapping from a symbol reference to the "common fragments" in its overload group.
+    static let commonFragmentsMap: Synchronized<[ResolvedTopicReference: [SymbolGraph.Symbol.DeclarationFragments.Fragment]]> = .init([:])
+
+    /// Set the common fragments for the given symbol references.
+    func setCommonFragments(
+        references: [ResolvedTopicReference],
+        fragments: [SymbolGraph.Symbol.DeclarationFragments.Fragment])
+    {
+        Self.commonFragmentsMap.sync({ map in
+            for reference in references {
+                map[reference] = fragments
+            }
+        })
+    }
+
+    /// Fetch the common fragments for the given reference, if present.
+    func commonFragments(
+        for reference: ResolvedTopicReference
+    ) -> [SymbolGraph.Symbol.DeclarationFragments.Fragment]? {
+        Self.commonFragmentsMap.sync({ $0[reference] })
+    }
+
+    /// Fetch the common fragments for the given references, or compute it if necessary.
+    func commonFragments(
+        for mainDeclaration: OverloadDeclaration,
+        overloadDeclarations: [OverloadDeclaration]
+    ) -> [SymbolGraph.Symbol.DeclarationFragments.Fragment] {
+        if let fragments = commonFragments(for: mainDeclaration.reference) {
```

Comment:
See the previous comment. If you only ever query the fragments for the main declaration here, then why save the same fragments for each of the overloads in the call to `setCommonFragments` below?